src/hardware/sblaster.cpp

```diff
@@ -186,16 +186,28 @@ struct SbInfo {
 	} dac = {};
 
 	struct {
-		uint8_t index     = 0;
+		uint8_t index = 0;
+
+		// If true, DOS software can programmatically change the Sound
+		// Blaster mixer's volume levels on SB Pro 1 and later card for
+		// the SB (DAC), OPL (FM) and CDAUDIO channels. These are called
+		// "app levels". The final output level is the "user level" set
+		// in the DOSBox mixer multiplied with the "app level" for these
+		// channels.
+		//
+		// If the Sound Blaster mixer is disabled, the "app levels"
+		// default to unity gain.
+		//
+		bool enabled = false;
+
+		uint8_t dac[2] = {}; // can be controlled programmatically
+		uint8_t fm[2]  = {}; // can be controlled programmatically
+		uint8_t cda[2] = {}; // can be controlled programmatically
```

Comment:
I don't really mind end of line comments too much personally but I know you complain about them so I'll mention it :smile:

Reply:
Haha, yeah, I avoid them in general for regular code, but in some special cases they can be OK. E.g., like here for struct member definitions to keep a tabular format.